README.md

```diff
@@ -1,12 +1,342 @@
+| Ember Data 1.0.0-beta19.x         | OK            |
+| Ember Data 1.13.x                 | OK            |
+| Ember Data 2.x.x                  | unknown       |
+
+## Writing decorators
+Put your decorators in `app/decorators` directory.
+Here is example of basic decorator:
+
+``` javascript
+import ModelDecorator from 'ember-clothier/model-decorator';
+
+export default ModelDecorator.extend({
+
+  // decorate model with full name attribute
+  fullName: function() {
+    return this.get('firstName') + this.get('lastName');
+  }.property('firstName', 'lastName'),
+
+  // handle state in menu
+  isActiveInMenu: false,
+
+  changeInMenu: function() {
+    this.toggleProperty('isActiveInMenu');
+  }
+});
+```
+
+## Aliases and Naming Conventions
+If you do not specify decorator name when decorating object default decorator will be used.
+With Ember Data this is done by using modelName key provided in `DS.Model` instance.
+If you are not using Ember Data you have to specify `_modelName` attribute in your model.
+The `_modelName` attribute should be also use for overwriting default name with Ember Data.
+
+project structure:
+```
+app/
+  |- decorators
+  |   |- user.js
+  |   |- menu-item.js
+  |- models
+  |   |- user.js
+  |- routes
+      |- application.js
+```
+
+```javascript
+// models/user.js
+import Ember from 'ember';
+import DecoratorMixin from 'ember-clothier/model-mixin';
+
+export default Ember.Object.extend(DecoratorMixin, {
+  _modelName: 'user', // this.decorate() will lookup for user decorator
+  name: ''
+});
+```
+
+Then in your route:
+```javascript
+import Ember from 'ember';
+import UserModel from '../models/user';
+
+export default Ember.Route.extend({
+  model: function() {
+    return UserModel.create({ name: 'Tom Dale' });
+  },
+  setupController: function(controller, model) {
+    this._super(controller, model);
+    controller.set('user', model.decorate()); // RETURNS MODEL DECORATED WITH USER DECORATOR
+    controller.set('menuItem', model.decorate('menu-item')); // RETURNS MODEL DECORATED WITH MENU-ITEM DECORATOR
+  }
+});
+```
+
+## Decorating Objects and Collections
+With Clothier it is simple to decorate both objects and collections.
+There are two basic mixins which implements methods for creating decorators instances.
+`ModelMixin` implements `decorate()` method for decorating model instances and helpers for decorating its relationships (see below).
+`RouteMixin` implements `decorateModel()` and `decorateCollection()` methods for both **objects and collections**.
+The difference is that Route methods takes two arguments where first one is model or collection of instances.
+See Api Documentation for more informations.
+
+## Decorating relationships
+`ModelMixin` comes with two additional helper functions for decorating model relationships.
+This helpers takes two arguments ­ *relationKey* and *decoratorName* and return `Ember.computed` which returns decorated relationship.
```

Comment:
The description "This helpers takes two arguments ­ *relationKey* and *decoratorName* and return ..." contradicts the example bellow - `decorateBelongsTo('author')`.

Also as I've mentioned in https://github.com/GlobalWebIndex/pro/pull/146#discussion_r40257728 the usage of model decorator without a context doesn't sound right. Mandatory decorator name parameter would remove such a confusion (at the potential cost of writing decorator name - which should actually be an indicator of decorator context)